ostd/src/arch/x86/task/switch.S

```diff
@@ -1,32 +1,35 @@
 /* SPDX-License-Identifier: MPL-2.0 */
 
 .text
-.global context_switch
 .code64
-context_switch: # (cur: *mut TaskContext, nxt: *TaskContext)
+.global context_switch
+.global first_context_switch
+
+context_switch: # (nxt: *const TaskContext, cur: *mut TaskContext)
   # Save cur's register
   mov rax, [rsp] # return address
-  mov [rdi + 56], rax # 56 = offsetof(TaskContext, rip)
-  mov [rdi + 0], rsp
-  mov [rdi + 8], rbx
-  mov [rdi + 16], rbp
-  mov [rdi + 24], r12
-  mov [rdi + 32], r13
-  mov [rdi + 40], r14
-  mov [rdi + 48], r15
+  mov [rsi + 56], rax # 56 = offsetof(TaskContext, rip)
+  mov [rsi + 0], rsp
+  mov [rsi + 8], rbx
+  mov [rsi + 16], rbp
+  mov [rsi + 24], r12
+  mov [rsi + 32], r13
+  mov [rsi + 40], r14
+  mov [rsi + 48], r15
   rdfsbase r15
-  mov [rdi + 64], r15 # 64 = offsetof(TaskContext, fsbase)
+  mov [rsi + 64], r15 # 64 = offsetof(TaskContext, fsbase)
 
+first_context_switch:
```

Comment:
I think we should remove the blank line before `first_context_switch` to highlight the fact that it is part of the `context_switch` function. A quick glimpse of the current code may get the wrong impression that the `context_switch` function already ends before `first_context_switch`.